src/botcity/maestro/sdk.ts

```diff
@@ -306,4 +306,74 @@ export class BotMaestroSdk {
     await fs.promises.writeFile(filepath, response.data)
     return response.data
   }
+
+  @ensureAccessToken
+  @catchError
+  async getCredential (
+    label: string,
+    key: string
+  ): Promise<any> {
+    const url = `${this._server}/api/v2/credential/${label}/key/${key}`
+    const response: AxiosResponse = await axios.get(url, this.headers).catch((error: any) => {
+      throw new Error(error.response.data.message)
+    })
+    return response.data
+  }
+
+  @ensureAccessToken
+  @catchError
+  async getCredentialByLabel (
+    label: string
+  ): Promise<any> {
+    const url = `${this._server}/api/v2/credential/${label}`
+    const response: AxiosResponse = await axios.get(url, this.headers).catch((error: any) => {
+      throw new Error(error.response.data.message)
+    })
+    return response.data
+  }
+
+  @ensureAccessToken
+  @catchError
+  async createCredentialByLabel (
+    label: string,
+    key: string,
+    value: any
+  ): Promise<any> {
+    const data = {
+      label,
+      secrets: [
+        { key, value, valid: true }
+      ]
+    }
+    const url = `${this._server}/api/v2/credential`
+    const credential = await axios.post(url, data, this.headers).catch((error: any) => {
+      throw new Error(error.response.data.message)
+    })
+    return credential
+  }
+
+  @ensureAccessToken
+  @catchError
+  async createCredential (
+    label: string,
+    key: string,
+    value: any
+  ): Promise<any> {
+    let credential
+    try {
+      credential = await this.getCredentialByLabel(label)
+      if (credential == null) throw new Error('Credential not found')
+      const url = `${this._server}/api/v2/credential/${label}/key`
+      const data = { key, value }
+      await axios.post(url, data, this.headers).catch((error: any) => {
+        throw new Error(error.response.data.message)
+      })
+    } catch (error) {
+      let message = ''
+      if (error instanceof Error) message = error.message
+      if (message.toLowerCase() !== 'credential not found') throw new Error(`Error during message. Server returned: ${message}`)
+      credential = await this.createCredentialByLabel(label, key, value)
+    }
```

Comment:
I believe this code block could be simplified. How about this?
```suggestion
      credential = await this.getCredentialByLabel(label)
      if (credential == null) {
          credential = await this.createCredentialByLabel(label, key, value)
          return credential
      }
      const url = `${this._server}/api/v2/credential/${label}/key`
      const data = { key, value }
      await axios.post(url, data, this.headers).catch((error: any) => {
        throw new Error(error.response.data.message)
      })
      credential = ????
```